### PR TITLE
Add tests for pluginRegistry

### LINKED ISSUE #1: Needs more tests
[Issuehunt badges](https://issuehunt.io/r/concordancejs/concordance/issues/1)

Code coverage is incomplete, which may be hiding bugs in the logic.

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Sponsors (Total: $40.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($40.00)

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/concordancejs/concordance/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

### PR DESCRIPTION
**Pull request by bunysae**
I added tests for `pluginRegistry` in order to fix #1.

### CODE REVIEW

**Comment by novemberborn** [comment]
Thanks for your PR @bunysae. I did not open my editor this weekend so I haven't had time yet to look into this, sorry.

**Comment by novemberborn** [comment]
So I don't remember much about how this code works, but I just did update the tooling and dependencies. Could you try rebasing? Apologies for any conflicts.

**Comment by novemberborn** [comment]
@bunysae I don't think that did the trick. All the changes are now showing up in the diff, and according to GitHub there's still merge conflicts.

**Comment by bunysae** [comment]
@novemberborn 
Fixed :). I rebased the wrong branch yesterday.

**Comment by novemberborn** [comment]
@bunysae cool, thanks. Looks like there are some linter errors now.

You can run `npx as-i-preach --fix` to fix.

File: test/pluginRegistry.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      return tryDescribeValue
+    },
+  }
+
+  const id2deserialize = new Map()
+  id2deserialize.set(1, 'objectDeserializer')
+  id2deserialize.set(2, 'arrayDeserializer')
+  const tag2id = new Map()
+  tag2id.set('complexCustomObject', 1)
+  tag2id.set('complexCustomArray', 2)
```
Comment (novemberborn) [inline test/pluginRegistry.js]: I'd prefer this syntax, which is more concise:

```js
const tag2id = new Map([['complexCustomObject', 1], ['complexCustomArray', 2]])
```

File: test/fixtures/customErrorPlugin.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  constructor (message, code) {
+    super(message)
+    this.code = code
+    this.name = 'CustomError'
+  }
+}
+
+exports.CustomError = CustomError
+
+exports.setDependencies = function (dependencies) {
```
Comment (novemberborn) [inline test/fixtures/customErrorPlugin.js]: I think it'd be better to follow the `register` pattern used here:

https://github.com/concordancejs/react/blob/cc84400d6f805fd57338f4101abd4e30d43f3508/index.js#L59

File: test/pluginRegistry.js (hunk 1 above)
Comment (bunysae) [reply test/pluginRegistry.js]: Is now not need anymore (moved to snapshot)

File: test/pluginRegistry.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  const expected = {
+    id2deserialize,
+    serializerVersion: 1,
+    name: 'complex',
+    tag2id,
+    theme: {},
+    tryDescribeValue,
+  }
+
+  t.deepEqual(pluginRegistry.add(plugin), expected)
```
[the remark this answers is not in the document]
Comment (bunysae) [reply test/pluginRegistry.js]: I moved it to a snapshot.

File: test/fixtures/customErrorPlugin.js (hunk 2 above)
Comment (bunysae) [reply test/fixtures/customErrorPlugin.js]: Done

**Comment by novemberborn** [comment]
Thank you @bunysae.

**Pull request merged**